src/main/java/com/rcv/ContestConfig.java

```diff
@@ -203,6 +213,21 @@ private void validateCvrFileSources() {
           Logger.log(Level.SEVERE, "CVR file not found: %s", cvrPath);
         }
 
+        // perform CDF checks
+        if (source.getProvider().equals("CDF")) {
+          if (rawConfig.cvrFileSources.size() != 1) {
+            isValid = false;
+            Logger.log(Level.SEVERE, "CDF files must be tabulated individually.");
+          }
+          if (isTabulateByPrecinctEnabled()) {
+            isValid = false;
+            Logger.log(Level.SEVERE, "Tabulate By Precinct may not be used with CDF files.");
+          }
+          continue;
```

Comment:
Instead of this, why not just put the ES&S checks in an `else` block?

Reply:
yeah that's more readable